### ISSUE TITLE
Empty build output

### ISSUE THREAD

**Issue by pierzchalski** [jeff-hykin/nix_version_search_cli#6]
The command in the README doesn't install anything:
```bash
$ nix profile install \
    'https://github.com/jeff-hykin/nix_version_search_cli/archive/423e829ee2f40e99be233de63f09d44dd39e9f67.tar.gz#nvs'
$ which nvs

$ ls ~/.nix-profile/bin/ | rg nvs

```

Originally I thought I was doing something wrong, but it looks like the `nix_version_search_cli` flake doesn't actually build anything:

```bash
$ nix build 'github:jeff-hykin/nix_version_search_cli?ref=v1.4.8#nvs'
$ tree result
result
└── bin

1 directory, 0 files
```

Any idea what's going on?

**Comment by jeff-hykin** [comment]
Thats really weird. I'm not sure how it could do that without at least failing to build. Thanks for reporting this!

**Comment by jeff-hykin** [comment]
Something strange is going on, and seems like a flakes-related issue.

This is my MacM1 is fine, even using flakes
<img width="1118" alt="Screen Shot 2024-06-28 at 7 15 52 PM" src="https://github.com/jeff-hykin/nix_version_search_cli/assets/17692058/fb3c87d0-aca8-42ea-98e7-a86c037e4da1">

My linux machine works for non-flakes (my default there) but gets an empty build for flakes
```sh

$ nix --version
nix (Nix) 2.17.0

$ which nvs

$ nix-env -i -f https://github.com/jeff-hykin/nix_version_search_cli/archive/423e829ee2f40e99be233de63f09d44dd39e9f67.tar.gz
installing 'nvs'
this derivation will be built:
  /nix/store/p1p4nq9abrhblsp155zf56lb1wzr44is-nvs.drv
these 14 paths will be fetched (35.39 MiB download, 135.27 MiB unpacked):
  /nix/store/d3zxgr5xjn057j9kbhd80gmjpiq80n60-acl-2.3.1
  /nix/store/jbjamv1l5zxg2lz2ij0ygbddv0yvyn21-attr-2.5.1
  /nix/store/lf0wpjrj8yx4gsmw2s3xfl58ixmqk8qa-bash-5.2-p15
  /nix/store/7vg2sgzv2hidmj8plg4w31x2sqbbkfrg-bzip2-1.0.8
  /nix/store/vwkvhj69z4qqgmpa2lwm97kabf12p26r-coreutils-9.3
  /nix/store/prfspc73h22gng0pk7nk9vf8q3csazlb-deno-1.37.2
  /nix/store/9fy9zzhf613xp0c3jsjxbjq6yp8afrsv-gcc-12.3.0-lib
  /nix/store/1q6qwq8csbhyy0pv54sab00jxlmb3rw2-gcc-12.3.0-libgcc
  /nix/store/gqghjch4p1s69sv4mcjksb2kb65rwqjy-glibc-2.38-23
  /nix/store/dk95bb4ihip42pssr6ygf90cljamzi7p-gmp-with-cxx-6.3.0
  /nix/store/w4djxksksd1p8m054k537plqnif5858k-libidn2-2.3.4
  /nix/store/4r64z7v5l40pg6r0hd169bcs85c8c42b-libunistring-1.1
  /nix/store/8ksjsax1pyzpk416g3n112nij5g5zzm8-unzip-6.0
  /nix/store/sdhmm56jv7r449hf7f57cl3pvnsy3cbg-xgcc-12.3.0-libgcc
copying path '/nix/store/1q6qwq8csbhyy0pv54sab00jxlmb3rw2-gcc-12.3.0-libgcc' from 'https://cache.nixos.org'...
copying path '/nix/store/sdhmm56jv7r449hf7f57cl3pvnsy3cbg-xgcc-12.3.0-libgcc' from 'https://cache.nixos.org'...
copying path '/nix/store/4r64z7v5l40pg6r0hd169bcs85c8c42b-libunistring-1.1' from 'https://cache.nixos.org'...
copying path '/nix/store/w4djxksksd1p8m054k537plqnif5858k-libidn2-2.3.4' from 'https://cache.nixos.org'...
copying path '/nix/store/gqghjch4p1s69sv4mcjksb2kb65rwqjy-glibc-2.38-23' from 'https://cache.nixos.org'...
copying path '/nix/store/jbjamv1l5zxg2lz2ij0ygbddv0yvyn21-attr-2.5.1' from 'https://cache.nixos.org'...
copying path '/nix/store/9fy9zzhf613xp0c3jsjxbjq6yp8afrsv-gcc-12.3.0-lib' from 'https://cache.nixos.org'...
copying path '/nix/store/lf0wpjrj8yx4gsmw2s3xfl58ixmqk8qa-bash-5.2-p15' from 'https://cache.nixos.org'...
copying path '/nix/store/7vg2sgzv2hidmj8plg4w31x2sqbbkfrg-bzip2-1.0.8' from 'https://cache.nixos.org'...
copying path '/nix/store/d3zxgr5xjn057j9kbhd80gmjpiq80n60-acl-2.3.1' from 'https://cache.nixos.org'...
copying path '/nix/store/8ksjsax1pyzpk416g3n112nij5g5zzm8-unzip-6.0' from 'https://cache.nixos.org'...
copying path '/nix/store/prfspc73h22gng0pk7nk9vf8q3csazlb-deno-1.37.2' from 'https://cache.nixos.org'...
copying path '/nix/store/dk95bb4ihip42pssr6ygf90cljamzi7p-gmp-with-cxx-6.3.0' from 'https://cache.nixos.org'...
copying path '/nix/store/vwkvhj69z4qqgmpa2lwm97kabf12p26r-coreutils-9.3' from 'https://cache.nixos.org'...
building '/nix/store/p1p4nq9abrhblsp155zf56lb1wzr44is-nvs.drv'...
building '/nix/store/cmawfbbikgnccq8ws0g5479k5cc3l0nx-user-environment.drv'...

$ which nvs
/home/jeffhykin/.nix-profile/bin/nvs

$ nix --extra-experimental-features nix-command --extra-experimental-features flakes build 'github:jeff-hykin/nix_version_sear
ch_cli?ref=v1.4.8#nvs'
[1/0/1 built, 15 copied (404.2 MiB), 96.3 MiB DL] building nvs

$ tree result/
result/
`-- bin

2 directories, 0 files
```


Edit: making progress but I'll have to finish later. Looks like during the build it gets this error on linux, but only when using `nix build` not `nix-build`. I can also confirm that URL/file exists.

Maybe its some kind of network sandboxing thing I'm unaware of with nix flakes?
```
[0m[1m[31merror[0m: Writing /nix/store/lm66dq0f4x9hhqbbvk0pxxsqlszbwjsv-nvs/bin/nvs

Caused by:
    0: error sending request for url (https://dl.deno.land/release/v1.44.4/denort-x86_64-unknown-linux-gnu.zip): error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
    1: error trying to connect: dns error: failed to lookup address information: Temporary failure in name resolution
    2: dns error: failed to lookup address information: Temporary failure in name resolution
    3: failed to lookup address information: Temporary failure in name resolution
```

**Comment by pierzchalski** [comment]
Fixed in v1.4.29
